packages/react/spec/auto/shadcn-defaults/components/Calendar.tsx

```diff
@@ -1,50 +1,409 @@
+  displayYears,
+  ...props
+}: {
+  showYearSwitcher?: boolean;
+  navView: NavView;
+  setNavView: React.Dispatch<React.SetStateAction<NavView>>;
+  displayYears: { from: number; to: number };
+} & React.HTMLAttributes<HTMLSpanElement>) {
+  if (!showYearSwitcher) return <span {...props}>{children}</span>;
+  return (
+    <Button
+      className="h-7 w-full truncate text-sm font-medium"
+      variant="ghost"
+      size="sm"
+      onClick={() => setNavView((prev) => (prev === "days" ? "years" : "days"))}
+    >
+      {navView === "days" ? children : displayYears.from + " - " + displayYears.to}
+    </Button>
+  );
+}
+
+function MonthGrid({
+  className,
+  children,
+  displayYears,
+  startMonth,
+  endMonth,
+  navView,
+  setNavView,
+  ...props
+}: {
+  className?: string;
+  children: React.ReactNode;
+  displayYears: { from: number; to: number };
+  startMonth?: Date;
+  endMonth?: Date;
+  navView: NavView;
+  setNavView: React.Dispatch<React.SetStateAction<NavView>>;
+} & React.TableHTMLAttributes<HTMLTableElement>) {
+  if (navView === "years") {
+    return (
+      <YearGrid
+        displayYears={displayYears}
+        startMonth={startMonth}
+        endMonth={endMonth}
+        setNavView={setNavView}
+        navView={navView}
+        className={className}
+        {...props}
+      />
+    );
+  }
+  return (
+    <table className={className} {...props}>
+      {children}
+    </table>
+  );
+}
+
+function YearGrid({
+  className,
+  displayYears,
+  startMonth,
+  endMonth,
+  setNavView,
+  navView,
+  ...props
+}: {
+  className?: string;
+  displayYears: { from: number; to: number };
+  startMonth?: Date;
+  endMonth?: Date;
+  setNavView: React.Dispatch<React.SetStateAction<NavView>>;
+  navView: NavView;
+} & React.HTMLAttributes<HTMLDivElement>) {
+  const { goToMonth, selected } = useDayPicker();
+
+  return (
+    <div className={cn("grid grid-cols-4 gap-y-2", className)} {...props}>
+      {Array.from({ length: displayYears.to - displayYears.from + 1 }, (_, i) => {
+        const isBefore = differenceInCalendarDays(new Date(displayYears.from + i, 11, 31), startMonth!) < 0;
+
+        const isAfter = differenceInCalendarDays(new Date(displayYears.from + i, 0, 0), endMonth!) > 0;
+
+        const isDisabled = isBefore || isAfter;
+        return (
+          <Button
+            key={i}
+            className={cn(
+              "h-7 w-full text-sm font-normal text-foreground",
+              displayYears.from + i === new Date().getFullYear() && "bg-accent font-medium text-accent-foreground"
+            )}
+            variant="ghost"
+            onClick={() => {
+              setNavView("days");
+              goToMonth(new Date(displayYears.from + i, (selected as Date | undefined)?.getMonth() ?? 0));
+            }}
+            disabled={navView === "years" ? isDisabled : undefined}
+          >
+            {displayYears.from + i}
+          </Button>
+        );
+      })}
+    </div>
+  );
+}
+
```

Comment:
when we go to integrate this new version into gadget we may need to refresh our component templates to drop this new one, FYI there is a `development/refresh-shadcn.mjs` script for that